pkg/alicloud/client/types.go

```diff
@@ -34,47 +38,99 @@ type ClientFactory interface {
 	NewSLBClient(region, accessKeyID, accessKeySecret string) (SLB, error)
 	NewVPCClient(region, accessKeyID, accessKeySecret string) (VPC, error)
 	NewRAMClient(region, accessKeyID, accessKeySecret string) (RAM, error)
-	NewStorageClientFromSecretRef(ctx context.Context, client client.Client, secretRef *corev1.SecretReference, region string) (Storage, error)
+	NewOSSClient(endpoint, accessKeyID, accessKeySecret string) (OSS, error)
+	NewOSSClientFromSecretRef(ctx context.Context, client client.Client, secretRef *corev1.SecretReference, region string) (OSS, error)
 }
 
-// STS is an interface which must be implemented by alicloud sts clients.
-type STS interface {
-	GetAccountIDFromCallerIdentity(ctx context.Context) (string, error)
+// ecsClient implements the ECS interface.
+type ecsClient struct {
+	ecs.Client
 }
 
-// ECS is an interface which must be implemented by alicloud ecs clients.
+// ECS is an interface which declares ECS related methods.
 type ECS interface {
 	CheckIfImageExists(ctx context.Context, imageID string) (bool, error)
 	ShareImageToAccount(ctx context.Context, regionID, imageID, accountID string) error
+	DescribeSecurityGroups(ctx context.Context, request *ecs.DescribeSecurityGroupsRequest) (response *ecs.DescribeSecurityGroupsResponse, err error)
+	DescribeSecurityGroupAttribute(ctx context.Context, request *ecs.DescribeSecurityGroupAttributeRequest) (response *ecs.DescribeSecurityGroupAttributeResponse, err error)
+	DescribeKeyPairs(ctx context.Context, request *ecs.DescribeKeyPairsRequest) (response *ecs.DescribeKeyPairsResponse, err error)
 }
 
-// SLB is an interface which must be implemented by alicloud slb clients.
+// stsClient implements the STS interface.
+type stsClient struct {
+	sts.Client
```

Comment:
We do  not need a wrapper struct. We just need to return an instance of `sts.Client` for those NewSTSClient. The same for other wrappers.

Reply:
stsClient as well as other wrappers, not only would be used to call interface provided by Alicloud, but also implements the interface which defined by us. So we need this kind of wrapper.